mobify/platform-scaffold
# Add AstroConnector component to allow interaction between webviews

To coordinate different webviews in the app we need to be able to notify them of changes that will affect the store, this is an initial implementation of a `NativeConnector` component that will serve to listen for/trigger notifications to the native parts of the app.

 **JIRA**: https://mobify.atlassian.net/browse/HYB-1025
 **Linked PRs**:  https://github.com/mobify/astro/pull/790

## Changes
- Move all app events to `AppEvents.js`
- Add `NativeConnector` component
- Connect app to native connector for events

## How to test-drive this PR
- Change `package.json` to point to linked PR
- Run `npm run deps` in the `native` folder
- Change `baseConfig.js` in `native/config` to enable preview if disabled
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Go to product
- Go back
- Go to product
- Add to cart
- Go to cart
- Remove from cart
- Close cart
- Go back
- Cart icon should not show a red dot with a 1
